src/com/couchbase/tools/performer/BuildDockerColumnarGoPerformer.groovy

```diff
@@ -0,0 +1,49 @@
+package com.couchbase.tools.performer
+
+import com.couchbase.context.environments.Environment
+import com.couchbase.tools.tags.TagProcessor
+import groovy.transform.CompileStatic
+
+import java.util.regex.Pattern
+
+@CompileStatic
+class BuildDockerColumnarGoPerformer {
+    private static final String VERSION_ARG = "SDK_VERSION"
+    private static final String GERRIT_REF_ARG = "SDK_GERRIT_REF"
+
+    /**
+     * @param path absolute path to above 'transactions-fit-performer'
+     * @param build what to build
+     */
+    static void build(Environment imp, String path, VersionToBuild build, String imageName, boolean onlySource = false, Map<String, String> dockerBuildArgs = [:]) {
+        imp.log("Building Columnar Go ${build}")
+
+        // Build context needs to be perf-sdk as we need the .proto files
+        imp.dirAbsolute(path) {
+            imp.dir('transactions-fit-performer') {
+                imp.dir('performers/columnar/go') {
+                    TagProcessor.processTags(new File(imp.currentDir()), build, Optional.of(Pattern.compile(".*\\.go")))
+                }
+
+                // We check HasSha and HasGerrit _before_ HasVersion, since BuildShaVersion & BuildGerritVersion
+                // implement HasVersion as well as HasSha/HasGerrit
+                if (build instanceof HasSha) {
+                    dockerBuildArgs.put(VERSION_ARG, build.sha())
+                }
+                else if (build instanceof HasVersion) {
+                    dockerBuildArgs.put(VERSION_ARG, "v${build.version()}".toString())
+                }
+                else if (build instanceof BuildMain) {
+                    dockerBuildArgs.put(VERSION_ARG, "master")
+                }
+
```

Comment:
NIT: I would add a branch for `build instanceof HasGerrit` that throws a `RuntimeException`. It's not strictly necessary, but might be good in case anyone attempts to do that.

Reply:
I'm a bit nervous of doing that in case it breaks a bunch of things.

Reply:
I don't think it will break anything, `HasGerrit` shouldn't be attempted anyway with Go columnar. This is how we handled this in the past. If there is no exception, it will silently build main, which is not ideal.

Reply:
Done